### PR TITLE
docs: std storage read terminator functions

### PR DESCRIPTION
**Pull request by shawnharmsen**
closes: https://github.com/foundry-rs/book/issues/317

If this looks right I can continue expanding std storage here:
https://book.getfoundry.sh/reference/forge-std/std-storage.html#functions

~target~
sig
with_key
depth
checked_write
find

### CODE REVIEW

File: src/reference/forge-std/read.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+## `read`
+
+### Signature
+
+```solidity
+function read(StdStorage storage self) private returns (bytes memory);
+```
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/read.md:7]: `private` function, can't access

**Comment by shawnharmsen** [comment]
Finished the remainder

1. unsure about find description
- copied over comments from here: https://github.com/foundry-rs/forge-std/blob/1680d7fb3e00b7b197a7336e7c88e838c7e6a3ec/src/Test.sol#L424-L430

File: src/reference/forge-std/with_key.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Examples
+
+```solidity
+uint256 slot = stdstore
+    .target(address(test))
+    .sig(test.deep_map.selector)
+    .with_key(address(this))
+    .with_key(address(this))
+    .find();
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/with_key.md:29]:
```suggestion
// mapping(uint256 => mapping(address => uint256)) public itemToPlayerToQuantity;

uint256 slot = stdstore
    .target(address(game))
    .sig(game.itemToOwnerToQuantity.selector)
    .with_key(1337)
    .with_key(address(this))
    .find();
```

File: src/reference/forge-std/target.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+### Examples
+
+```solidity
+uint256 slot = stdstore
+    .target(address(test))
+    .sig(test.map_struct.selector)
+    .with_key(address(this))
+    .depth(0)
+    .find();
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/target.md:21]:
```suggestion
// function playerCount() public view returns (uint256) {

uint256 slot = stdstore
    .target(address(game))
    .sig(game.playerCount.selector)
    .find();
```

File: src/reference/forge-std/sig.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+### Examples
+
+```solidity
+uint256 slot = stdstore
+    .target(address(test))
+    .sig(test.map_struct.selector)
+    .with_key(address(this))
+    .depth(1)
+    .find();
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/sig.md:25]:
```suggestion
// function playerCount() public view returns (uint256) {

uint256 slot = stdstore
    .target(address(game))
    .sig(game.playerCount.selector)
    .find();
```

File: src/reference/forge-std/find.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+If deep map, will be 
+```solidity
+keccak256(abi.encode(key1, keccak256(abi.encode(key0, uint(slot)))));
+```  
+
+If map struct, will be 
+```solidity
+bytes32(uint256(keccak256(abi.encode(key1, keccak256(abi.encode(key0, uint(slot)))))) + structFieldDepth);
+```  
+
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/find.md:32]:
```suggestion
```

File: src/reference/forge-std/find.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+### Examples
+
+```solidity
+uint256 slot = stdstore
+    .target(address(test))
+    .sig(test.deep_map_struct.selector)
+    .with_key(address(this))
+    .with_key(address(this))
+    .depth(1)
+    .find();
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/find.md:42]:
```suggestion
// uint256 public season;

uint256 slot = stdstore
    .target(address(game))
    .sig(game.season.selector)
    .find();
```

File: src/reference/forge-std/depth.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+### Examples
+
+```solidity
+uint256 slot = stdstore
+    .target(address(test))
+    .sig(test.deep_map_struct.selector)
+    .with_key(address(this))
+    .with_key(address(this))
+    .depth(0)
+    .find();
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/depth.md:22]:
```suggestion
// struct Character {
//     Class class;
//     uint256 level;
//     uint256 xp;
// }
//
// mapping (address => Character) public playerToCharacter;

stdstore
    .target(game)
    .sig(game.playerToCharacter.selector)
    .with_key(address(this))
    .depth(1)
    .checked_write(120);
```

File: src/reference/forge-std/checked_write.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+### Examples
+
+```solidity
+stdstore
+    .target(address(test))
+    .sig(test.deep_map.selector)
+    .with_key(address(this))
+    .with_key(address(this))
+    .checked_write(100);
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/checked_write.md:33]:
```suggestion
// uint256 public season;

uint256 slot = stdstore
    .target(address(game))
    .sig(game.season.selector)
    .checked_write(7);
```

**Review by ZeroEkkusu — COMMENTED** [review]
Suggestions to simplify some examples and make them more illustrative. Feel free to tweak them ofc!

**Review by onbjerg — APPROVED** [review]

**Pull request merged**